Editor/AGS.Editor/Entities/EditorPreferences.cs

```diff
@@ -242,6 +298,37 @@ private bool GetSettingsFromRegistry()
             return success;
         }
 
+        private List<PropertyInfo> GetBrowsableProperties()
+        {
+            return GetType().GetProperties()
+               .Where(f => f.GetCustomAttributes<BrowsableAttribute>().Contains(BrowsableAttribute.Yes))
+               .ToList();
+        }
+
+        public AppSettings CloneAppSettings()

```

Comment:
I don't get what the purpose of this is? I haven't read the entire conversation all the way back to the issue mind you, but ideally I shouldn't have to go anywhere outside of the code I'm looking at to understand it in the first place.

What confuses me is that I have to assume that we clone `AppSettings`  so that we can undo any changes made by a user if they hit cancel. But this use case is already covered in `ApplicationSettingsBase` with the `Reset()` and `Save()` isn't it?

Reply:
I don't know, [`ApplicationSettingsBase.Reset()`](https://docs.microsoft.com/en-us/dotnet/api/system.configuration.applicationsettingsbase.reset?view=netframework-4.6.2) has the following text:
> Restores the persisted application settings values to their corresponding default properties.

Is this wrong and it actually reloads the setting?